extras/drift_postgres/lib/src/pg_database.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:drift/backends.dart';
+import 'package:postgres/postgres.dart';
+
+///
+class PgDatabase extends DelegatedDatabase {
+  ///
+  PgDatabase(PostgreSQLConnection connection)
+      : super(_PgDelegate(connection, connection),
+            isSequential: true, logStatements: true);
+
+  ///
+  factory PgDatabase.open(String host, int port, String databaseName,
+      {String? username,
+      String? password,
+      int timeoutInSeconds = 30,
+      int queryTimeoutInSeconds = 30,
+      String timeZone = 'UTC',
+      bool useSSL = false,
+      bool isUnixSocket = false}) {
+    return PgDatabase(PostgreSQLConnection(host, port, databaseName,
+        username: username,
+        password: password,
+        timeoutInSeconds: timeoutInSeconds,
+        queryTimeoutInSeconds: queryTimeoutInSeconds,
+        timeZone: timeZone,
+        useSSL: useSSL,
+        isUnixSocket: isUnixSocket));
+  }
+}
+
+///
+class _PgDelegate extends DatabaseDelegate {
+  final PostgreSQLConnection _db;
+  final PostgreSQLExecutionContext _ec;
+
+  @override
+  SqlDialect get dialect => SqlDialect.postgres;
+
+  _PgDelegate(this._db, this._ec) : closeUnderlyingWhenClosed = false;
+
+  bool _isOpen = false;
+
+  final bool closeUnderlyingWhenClosed;
+
+  @override
+  TransactionDelegate get transactionDelegate => _PgTransactionDelegate(_db);
+
+  @override
+  late DbVersionDelegate versionDelegate;
+
+  @override
+  Future<bool> get isOpen => Future.value(_isOpen);
+
+  @override
+  Future<void> open(QueryExecutorUser user) async {
+    final pgVersionDelegate = _PgVersionDelegate(_db);
+
+    await _db.open();
+    await pgVersionDelegate.init();
+    await _initializeDatabase();
+
+    versionDelegate = pgVersionDelegate;
+    _isOpen = true;
+  }
+
+  Future _ensureOpen() async {
+    if (_db.isClosed) {
+      await _db.open();
+    }
+  }
+
+  Future<void> _initializeDatabase() async {
+    // TODO: Do we need create these functions?
+    //setup?.call(_db);
+  }
```

Comment:
I think this method can indeed be removed :D

Reply:
There is a 'useNativeFunctions()' call in here with native SQLite that creates custom functions. This functions is required? `regexp_moor_ffi, moor_contains, sqrt, etc.`

Reply:
It can be removed for postgres since these SQL functions are only available with a `NativeDatabase` (something that we explain in the docs too). We can't register custom functions on postgres so it should be fine.